airbyte-integrations/connectors/source-postgres/src/main/kotlin/io/airbyte/integrations/source/postgres/operations/types/InfFieldType.kt

```diff
@@ -0,0 +1,73 @@
+package io.airbyte.integrations.source.postgres.operations.types
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.NullNode
+import io.airbyte.cdk.data.AirbyteSchemaType
+import io.airbyte.cdk.data.JsonEncoder
+import io.airbyte.cdk.data.TextCodec
+import io.airbyte.cdk.jdbc.JdbcFieldType
+import io.airbyte.cdk.jdbc.JdbcGetter
+import java.sql.ResultSet
+
+// A generic field type which preserves infinity, -infinity and NaN values from DB to JSON
+class InfFieldType<T>(
+    airbyteSchemaType: AirbyteSchemaType,
+    baseJdbcGetter: JdbcGetter<T>,
+    baseJsonEncoder: JsonEncoder<T>,
+) :
+    JdbcFieldType<InfWrapper<T>>(
+        airbyteSchemaType,
+        InfJdbcGetter(baseJdbcGetter),
+        InfJsonEncoder(baseJsonEncoder),
+    )
+
+class InfJdbcGetter<T>(private val base: JdbcGetter<T>) : JdbcGetter<InfWrapper<T>> {
+    override fun get(rs: ResultSet, colIdx: Int): InfWrapper<T>? {
+        return InfWrapper.make(rs, colIdx, base)
+    }
+}
+
+class InfJsonEncoder<T>(private val base: JsonEncoder<T>) : JsonEncoder<InfWrapper<T>> {
```

Comment:
Since we're using the type of JsonEncoder as a hint for protobuf encoding, this may create a difficulty because the inner type is erased. 
Maybe `reified` can help here.
I'm noting this as a difficulty to myself.